ansible/roles/horizon-provision/tasks/main.yml

```diff
@@ -36,13 +36,13 @@
       Content-Type: "application/xml"
     body: "{{lookup('template','foreignSources_xml.j2')}}"
   no_log: true
-  register: requisitions_created
+  register: requisition_created
   when: requisition_exists
   with_dict: "{{ onms_policies }}"
 
 - name: "Verify if nodes exist in requisitions"
   ansible.builtin.uri:
-    url: "{{ onms_hzn.base_rest_url }}/{{ onms_hzn.requisitions_api }}/{{ onms_requisition }}/{{ onms_hzn.node_api }}/{{ ansible_facts['machine_id'] }}"
+    url: "{{ onms_hzn.base_rest_url }}/{{ onms_hzn.requisitions_api }}/{{ onms_requisition }}/{{ onms_hzn.node_api }}/{{ ansible_facts['machine_id'] if ansible_facts['machine_id'] is defined else inventory_hostname }}"
```

Comment:
`machine_id` is not available on nodes that don't get facts. We take the `inventory_hostname` here which is defined in the inventory.